Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseServiceClient.cs

```diff
@@ -0,0 +1,228 @@
+                                                   Name = @"\ApplicationInsights\Dependency Calls/Sec",
+                                                   Value = Round(sample.AIDependencyCallsPerSecond),
+                                                   Weight = 1
+                                               },
+                                           new MetricPoint
+                                               {
+                                                   Name = @"\ApplicationInsights\Dependency Call Duration",
+                                                   Value = Round(sample.AIDependencyCallDurationAveInMs),
+                                                   Weight = sample.AIDependencyCalls
+                                               },
+                                           new MetricPoint
+                                               {
+                                                   Name = @"\ApplicationInsights\Dependency Calls Failed/Sec",
+                                                   Value = Round(sample.AIDependencyCallsFailedPerSecond),
+                                                   Weight = 1
+                                               },
+                                           new MetricPoint
+                                               {
+                                                   Name = @"\ApplicationInsights\Dependency Calls Succeeded/Sec",
+                                                   Value = Round(sample.AIDependencyCallsSucceededPerSecond),
+                                                   Weight = 1
+                                               },
+                                            new MetricPoint
+                                               {
+                                                   Name = @"\ApplicationInsights\Exceptions/Sec",
+                                                   Value = Round(sample.AIExceptionsPerSecond),
+                                                   Weight = 1
+                                               }
+                                       };
+
+                metricPoints.AddRange(sample.PerfCountersLookup.Select(counter => new MetricPoint { Name = counter.Key, Value = Round(counter.Value), Weight = 1 }));
+
+                var dataPoint = new MonitoringDataPoint
+                                    {
+                                        Version = this.version,
+                                        InstrumentationKey = instrumentationKey,
+                                        Instance = this.instanceName,
+                                        Timestamp = sample.EndTimestamp,
+                                        Metrics = metricPoints.ToArray()
+                                    };
+
+                monitoringPoints.Add(dataPoint);
+            }
+
+            var serializer = new DataContractJsonSerializer(typeof(MonitoringDataPoint[]));
```

Comment:
Doesn't look like DataContractJsonSerializer is caching anything. Hence instantiating it every time.

I couldn't find anything that would point to potential security issues with this class (there are some reports of issues when reflection is disabled through CAS, but nothing else). Do you think testing it under the conditions that you stated above would be a sufficient verification?